### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by elliot710**

### CODE REVIEW

File: cucumber.json [hunk 1]
```diff
@@ -2,17 +2,30 @@
     "default": {
         "dryRun": false,
         "formatOptions": {
-            "snippetInterface": "async-await"
+            "snippetInterface": "aysnc-await"
```
Comment (bchmura-af) [inline cucumber.json:5]: typo?

File: e2e_tests/tests/steps/homePage.spec.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+let homePage:HomePage;
+
+
+Given('User is on home page', async function () {
+    this.attach(`User is on home page....!`);
+    homePage=new HomePage(getPage(), this.attach);
+});
+
+When('User searches for {string}', async function (string) {
+    homePage=new HomePage(getPage(), this.attach);
```
Comment (bchmura-af) [inline e2e_tests/tests/steps/homePage.spec.ts:20]: why create the HomePage again?

File: e2e_tests/tests/steps/homePage.spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    await homePage.searchForProduct(string);
+    await homePage.sortResults();
+    expect(await homePage.getSearchResults()).toBeTruthy(); //null check
+    
+});
+
+Then('The product {string} is found and added to the cart', async function (string) {
+    homePage=new HomePage(getPage(), this.attach);
+
```
Comment (bchmura-af) [inline e2e_tests/tests/steps/homePage.spec.ts:33]: no further calls in this test step?

File: e2e_tests/tests/steps/homePage.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+});
+
+When('User searches again for {string}', async function (string) {
+
```
Comment (bchmura-af) [inline e2e_tests/tests/steps/homePage.spec.ts]: empty test steps?

File: e2e_tests/tests/features/homepage.feature [hunk 5]
```diff
@@ -8,9 +8,9 @@ Feature: Search and Purchase on Amazon
     Given User is on home page
     When User searches for "Snickers"
     Then The product "Snickers" is found and added to the cart
-    And Fail last step intentionally
 
-#   Scenario: Search for Skittles and add to cart
-#     Given User is on home page
-#     When User searches again for "Skittles"
-#     Then The product "Skittles" is successfully found and added to the cart
+  Scenario: Search for Skittles and add to cart
```
Comment (bchmura-af) [inline e2e_tests/tests/features/homepage.feature:12]: Wasn't the task formulation: "A Cucumber-based descriptive test case to search for **the cheapest Snickers and Skittles**, add them to the basket, and verify the basket calculation and registration page redirection." ?

File: e2e_tests/tests/steps/homePage.spec.ts (hunk 3 above)
Comment (bchmura-af) [reply e2e_tests/tests/steps/homePage.spec.ts:33]: Ok, I see the steps were added in consecutive commits

File: cucumber.json (hunk 1 above)
Comment (elliot710) [reply cucumber.json:5]: oh, yeah. i bet there are more(

File: e2e_tests/tests/features/homepage.feature (hunk 5 above)
Comment (elliot710) [reply e2e_tests/tests/features/homepage.feature:12]: Yes, but I assumed it would be better splitting into two distinct ones. rewrite it?

Comment (bchmura-af) [reply e2e_tests/tests/features/homepage.feature:12]: No need to rewrite it.
However there is (purposefully) added difficulty when both are supposed to be in the same cart.

**Pull request merged**